common-arcana.lic

```diff
@@ -46,23 +46,42 @@ module DRCA
 
   def start_khris(khris, settings)
     khris
-      .map { |name| "Khri #{name.capitalize}" }
-      .each do |name|
-        activate_khri?(settings.kneel_khri, name)
+      .each do |khri_set|
+        activate_khri?(settings.kneel_khri, khri_set)
       end
   end
 
+  # Needs to handle (based on current usage):
+  # - Hasten
+  # - Delay Hasten
+  # - Hasten Focus
+  # - Delay Hasten Focus
+  # - Khri Hasten
+  # - Khri Delay Hasten
+  # - Khri Delay Hasten Focus
   def activate_khri?(settings_kneel, ability)
-    return false if DRSpells.active_spells[ability]
+    abilities = ability.split.map(&:capitalize)
 
-    kneel = kneel_for_khri?(settings_kneel, ability)
+    # Standardize for with/without 'Khri' on the front
+    abilities = abilities.drop(1) if abilities.first.casecmp('khri') == 0
+
+    # Handling for 'Delay'
+    should_delay = abilities.first.casecmp('delay') == 0
+    abilities = abilities.drop(1) if should_delay
```

Comment:
Lines 63-70 may make as a good utility method. I could see this being re-used in `combat-trainer` at line 2062 to split out the khris specified in the `buff_nonspells.khri` setting, particularly if people specify "khri delay xxx" or "khri xxx yyy zzz".

Reply:
the lines from buff_nonspells.khri end up passing through this code anyway.

The .map in combat_trainer that pins 'khri' on the front of everything could probably be removed as it shouldn't be needed for the timer key, and isn't needed as an input to this method anymore.